compiler/rustc_mir/src/transform/early_otherwise_branch.rs

```diff
@@ -250,12 +250,6 @@ impl<'a, 'tcx> Helper<'a, 'tcx> {
                 return None;
             }
 
-            // check that the value being matched on is the same. The
-            if this_bb_discr_info.targets_with_values.iter().find(|x| x.0 == value).is_none() {
```

Comment:
Where is this check duplicated? The check above does not look at `targets_with_values` at all.